src/EffectableEntity.js

```diff
@@ -0,0 +1,290 @@
+'use strict';
+
+const EventEmitter = require('events');
+const EffectList = require('./EffectList');
+const Attributes = require('./Attributes');
+
+/**
+ * @ignore
+ * @exports MetadatableFn
+ * @param {*} parentClass
+ * @return {module:MetadatableFn~Metadatable}
+ */
+class EffectableEntity extends EventEmitter
+{
+  constructor(data) {
+    super();
+
+    this.attributes = data.attributes || new Attributes();
+    this.effects = new EffectList(this, data.effects);
+  }
+
+  /**
+   * Proxy all events on the entity to effects
+   * @param {string} event
+   * @param {...*}   args
+   */
+  emit(event, ...args) {
+    super.emit(event, ...args);
+
+    this.effects.emit(event, ...args);
+  }
+
+  /**
+   * @param {string} attr Attribute name
+   * @return {boolean}
+   */
+  hasAttribute(attr) {
```

Comment:
This is an excellent refactor. Having attributes, effects, and damage possible on everything is huge.